build.mjs

```diff
@@ -2,10 +2,35 @@ import { readFile, writeFile, readdir } from "fs/promises";
 import { createHash } from "crypto";
 
 import { rollup } from "rollup";
-import { swc } from "rollup-plugin-swc3";
 import esbuild from "rollup-plugin-esbuild";
 import commonjs from "@rollup/plugin-commonjs";
 import nodeResolve from "@rollup/plugin-node-resolve";
+import swc from "@swc/core";
+
+const plugins = [
+    nodeResolve(),
+    commonjs(),
+    {
+        name: "swc",
+        async transform(code, id) {
+            const result = await swc.transform(code, {
+                filename: id,
+                jsc: {
+                    externalHelpers: true,
+                },
```

Comment:
You could pass
```js
parser: {
   syntax: "typescript",
   tsx: true
},
```
here so it always assumes the code is TSX, fixing issues with `.jsx` files etc., or you can add an extra check of the file name to make it parse accordingly

Reply:
If that had worked, I would have done it already. Also I pass SWC the file name, it infers by itself based on that.